DnDCharacterSheet/Interfaces/ICoinService.cs

```diff
@@ -0,0 +1,13 @@
+﻿using DTOs;
+
+namespace Interfaces
+{
+    public interface ICoinService
+    {
+        IEnumerable<CoinDTO> GetAllCoins();
+        CoinDTO AddCoin(CoinRequestDTO request);
+        CoinDTO GetCoinById(long id);
+        CoinDTO UpdateCoin(long id, CoinRequestDTO request);
+        bool DeleteCoin(long id);
+    }
+}
```

Comment:
Important, i don't think Coin term is correct, what i think you are refering to are Currencies